travis/04-run-localtests.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+source travis/00-_common.sh
+
+tests_path=$PWD/localtests
+test_logfile=/tmp/gh-ost-test.log
+exec_command_file=/tmp/gh-ost-test.bash
+
+echo_dot() {
+  echo -n "."
+}
+
+test_single() {
+  local test_name
+  test_name="$1"
+
+  echo -n "Testing: $test_name"
+
+  echo_dot
+  $SLAVE_MYSQL -e "stop slave; start slave; do sleep(1)"
+  echo_dot
+  $MASTER_MYSQL --default-character-set=utf8mb4 test < $tests_path/$test_name/create.sql
+
+  extra_args=""
+  if [ -f $tests_path/$test_name/extra_args ] ; then
+    extra_args=$(cat $tests_path/$test_name/extra_args)
+  fi
+  orig_columns="*"
+  ghost_columns="*"
+  order_by=""
+  if [ -f $tests_path/$test_name/orig_columns ] ; then
+    orig_columns=$(cat $tests_path/$test_name/orig_columns)
+  fi
+  if [ -f $tests_path/$test_name/ghost_columns ] ; then
+    ghost_columns=$(cat $tests_path/$test_name/ghost_columns)
+  fi
+  if [ -f $tests_path/$test_name/order_by ] ; then
+    order_by="order by $(cat $tests_path/$test_name/order_by)"
+  fi
+  # graceful sleep for replica to catch up
+  echo_dot
+  sleep 1
+  #
+  cmd="$GHOST_BIN \
+    --user=gh-ost \
+    --password=gh-ost \
+    --host=localhost \
+    --port=20002 \
+    --database=test \
+    --table=gh_ost_test \
+    --alter='engine=innodb' \
+    --exact-rowcount \
+    --switch-to-rbr \
+    --initially-drop-old-table \
+    --initially-drop-ghost-table \
+    --throttle-query='select timestampdiff(second, min(last_update), now()) < 5 from _gh_ost_test_ghc' \
+    --serve-socket-file=/tmp/gh-ost.test.sock \
+    --initially-drop-socket-file \
+    --postpone-cut-over-flag-file=/tmp/gh-ost.test.postpone.flag \
+    --test-on-replica \
+    --default-retries=1 \
+    --verbose \
+    --debug \
+    --stack \
+    --execute ${extra_args[@]}"
+  echo_dot
+  echo $cmd > $exec_command_file
+  echo_dot
+  bash $exec_command_file 1> $test_logfile 2>&1
+
+  execution_result=$?
```

Comment:
This is 0 even tho we have the error from the replication setup in another goroutine.